File: dotnet/windntrees.core/DataAccess.Core/Accounting/Transaction.cs
using System.ComponentModel.DataAnnotations;
using Microsoft.AspNetCore.Mvc;
using SharedLibrary.Core.Attributes;

namespace DataAccess.Core.Poultry
{
    [ModelMetadataType(typeof(TransactionMetaData))]
    public partial class Transaction
    {

    }

    public partial class TransactionMetaData
    {
        [Key]
        public System.Guid UID { get; set; }

        [LocaleMessageRequired]
        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Accounting.Transaction), Name = "TransactionTime")]
        public System.DateTime TransactionTime { get; set; }

        
        [LocaleMessageRequired]
        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Accounting.Transaction), Name = "UpdateTime")]
        public System.DateTime UpdateTime { get; set; }

        
        [LocaleMessageStringLength(100)]
        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Accounting.Transaction), Name = "ReferenceID")]
        public string ReferenceID { get; set; }

        
        [LocaleMessageStringLength(100)]
        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Accounting.Transaction), Name = "Reference")]
        public string Reference { get; set; }

        
        [LocaleMessageStringLength(100)]
        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Accounting.Transaction), Name = "CompositeID")]
        public string CompositeID { get; set; }

        
        [LocaleMessageRequired]
        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Accounting.Transaction), Name = "AccountID")]
        public System.Guid AccountID { get; set; }

        [LocaleMessageRequired]
        [LocaleMessageStringLength(20)]
        public string AccountNo { get; set; }

        [LocaleMessageRequired]
        [LocaleMessageStringLength(50)]
        public string Title { get; set; }

        [LocaleMessageRequired]
        [LocaleMessageStringLength(1000)]
        public string Particulars { get; set; }

        [LocaleMessageRequired]
        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Accounting.Transaction), Name = "DrAmount")]
        public decimal DrAmount { get; set; }

        [LocaleMessageRequired]
        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Accounting.Transaction), Name = "CrAmount")]
        public decimal CrAmount { get; set; }

        [LocaleMessageRequired]
        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Accounting.Transaction), Name = "Quantity")]
        public int Quantity { get; set; }

        [LocaleMessageRequired]
        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Accounting.Transaction), Name = "Active")]
        public bool Active { get; set; }

        
        [LocaleMessageStringLength(256)]
        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Accounting.Transaction), Name = "Username")]
        public string Username { get; set; }

        [Timestamp]
        public byte[] RowVersion { get; set; }
    }
}

File: dotnet/windntrees.core/Application.Core/wwwroot/js/models/Accounting/Transaction.js
function Transaction(data) {
    var instance = this;

    instance._datakey = data.Uid;

    instance.Uid = data.Uid;

    instance.TransactionTime = ko.observable(data.TransactionTime);

    instance.CompositeId = ko.observable((data.CompositeId === null || data.CompositeId === undefined) ? (moment(moment.utc(new Date()).toDate()).format('YYYY-MM-DD') + "-Transaction1") : data.CompositeId);

    instance.ReferenceId = ko.observable(data.ReferenceId);
    instance.ReferenceId.extend({
        maxLength: 100
    });

    instance.Reference = ko.observable(data.Reference);
    instance.Reference.extend({
        maxLength: 100
    });
    
    instance.AccountId = ko.observable(data.AccountId);
    instance.AccountId.extend({
        required: true
    });

    instance.AccountNo = ko.observable(data.AccountNo);
    instance.AccountNo.extend({
        required: true,
        maxLength: 20
    });

    instance.Title = ko.observable(data.Title);
    instance.Title.extend({
        required: true,
        maxLength: 50
    });

    instance.Particulars = ko.observable(data.Particulars);    
    instance.Particulars.extend({
        required: true,
        maxLength: 1000
    });

    instance.DrAmount = ko.observable(data.DrAmount);
    instance.DrAmount.extend({
        required: true,
        maxLength: 10
    });

    instance.CrAmount = ko.observable(data.CrAmount);
    instance.CrAmount.extend({
        required: true,
        maxLength: 10
    });

    instance.Quantity = ko.observable((data.Quantity !== null && data.Quantity !== undefined) ? data.Quantity : 1);
    instance.Quantity.extend({
        required: true
    });

    instance.Active = ko.observable(data.Active);
    instance.Username = ko.observable(data.Username);

    instance.LocalTransactionTime = ko.computed(function() {

        return new DateParser().parseDate(instance.TransactionTime(), true).toString('yyyy-MM-dd HH:mm:ss');
    });

    instance.RowVersion = data.RowVersion;
}